### PR TITLE
Feature/profile provider

### PR DESCRIPTION
**Pull request by Degorn**
Screenshot_1589271812

### CODE REVIEW

File: ecommers/i18n/en-US.json [hunk 1]
```diff
@@ -109,5 +109,12 @@
         "description": "Password restore link sent",
         "primary_button": "OK"
     },
-    "empty": ""
```
Comment (Platton96) [inline ecommers/i18n/en-US.json:125]: empty should be last

File: ecommers/lib/core/provider_models/profile_provider_model.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import 'package:ecommers/core/models/edit_profile_model.dart';
+import 'package:ecommers/shared/dependency_service.dart';
+import 'package:flutter/cupertino.dart';
+
+class ProfileProviderModel extends ChangeNotifier {
+  String getEmail() {
```
Comment (Platton96) [inline ecommers/lib/core/provider_models/profile_provider_model.dart]: use getter for example String get email =>profileService.user.email;

File: ecommers/lib/ui/pages/profile_edit_page.dart [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import 'package:ecommers/ui/widgets/button/primary_button_widget.dart';
+import 'package:ecommers/ui/widgets/index.dart';
+import 'package:ecommers/ui/widgets/profile/profile_image.dart';
+import 'package:flutter/material.dart' hide BackButton;
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:provider/provider.dart';
+
+class ProfileEditPage extends StatelessWidget {
+  static const EdgeInsets _listContainerMargin =
+      EdgeInsets.symmetric(horizontal: Insets.x5);
```
Comment (Platton96) [inline ecommers/lib/ui/pages/profile_edit_page.dart]: We decide to don't create consts for margins. You don't reuse  value

File: ecommers/lib/ui/pages/profile_edit_page.dart [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  final TextEditingController nameTextController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    final profileProvider =
+        Provider.of<ProfileProviderModel>(context, listen: false);
+    final passwordProvider =
+        Provider.of<ResetPasswordProviderModel>(context, listen: false);
+
+    nameTextController.text = profileProvider.getName();
```
Comment (Platton96) [inline ecommers/lib/ui/pages/profile_edit_page.dart]: getName should be getter

File: ecommers/lib/ui/pages/profile_edit_page.dart [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          borderSide: BorderSide(color: BrandingColors.backgroundIcon),
+        ),
+        fillColor: Colors.transparent,
+        filled: true,
+      ),
+    );
+  }
+
+  void _saveChanges(ProfileProviderModel profileProvider) {
+    final model = EditProfileModel(nameTextController.text);
```
Comment (Platton96) [inline ecommers/lib/ui/pages/profile_edit_page.dart]: move this method to provider

File: ecommers/lib/ui/pages/profile_edit_page.dart [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+                  profileProvider.getEmailOrPhone(),
+                  style: textTheme.bodyText2,
+                ),
+                const SizedBox(height: Insets.x3),
+                _buildTextFormField(nameTextController, localization.username),
+                const SizedBox(height: Insets.x5),
+                PrimaryButtonWidget(
+                  text: localization.reset_password,
+                  assetIconPath: Assets.arrowRightIcon,
+                  onPressedFunction: () => _resetPassword(
```
Comment (Platton96) [inline ecommers/lib/ui/pages/profile_edit_page.dart]: just call method provider

File: ecommers/lib/ui/pages/profile_edit_page.dart [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  Widget _buildProfileImage() {
+    return Stack(
+      children: [
+        const ProfileImage(Insets.x25),
+        Positioned(
+          right: 0,
+          top: 0,
+          child: Container(
+            alignment: Alignment.center,
+            height: Insets.x8,
```
Comment (Platton96) [inline ecommers/lib/ui/pages/profile_edit_page.dart]: Insets we use for padding margin

File: ecommers/i18n/en-US.json (hunk 1 above)
Comment (Degorn) [reply ecommers/i18n/en-US.json:125]: Moved

File: ecommers/lib/core/provider_models/profile_provider_model.dart (hunk 2 above)
Comment (Degorn) [reply ecommers/lib/core/provider_models/profile_provider_model.dart]: Done

File: ecommers/lib/ui/pages/profile_edit_page.dart (hunk 3 above)
Comment (Degorn) [reply ecommers/lib/ui/pages/profile_edit_page.dart]: Moved into body

File: ecommers/lib/ui/pages/profile_edit_page.dart (hunk 4 above)
Comment (Degorn) [reply ecommers/lib/ui/pages/profile_edit_page.dart]: Fixed

File: ecommers/lib/ui/pages/profile_edit_page.dart (hunk 5 above)
Comment (Degorn) [reply ecommers/lib/ui/pages/profile_edit_page.dart]: Moved

File: ecommers/lib/ui/pages/profile_edit_page.dart (hunk 7 above)
Comment (Degorn) [reply ecommers/lib/ui/pages/profile_edit_page.dart]: Replaced

File: ecommers/lib/ui/pages/profile_edit_page.dart (hunk 6 above)
Comment (Degorn) [reply ecommers/lib/ui/pages/profile_edit_page.dart]: Done

File: ecommers/lib/ui/pages/profile_page.dart [hunk 8]
```diff
@@ -52,63 +55,56 @@ class ProfilePage extends StatelessWidget {
   static const EdgeInsets _listContainerMargin =
       EdgeInsets.symmetric(horizontal: Insets.x5);
 
-  static const double _profileCardHeight = 100.0;
+  static const double _profileCardHeight = Insets.x25;
```
Comment (Platton96) [inline ecommers/lib/ui/pages/profile_page.dart]: remove variable, use just value

File: ecommers/lib/ui/pages/profile_page.dart [hunk 9]
```diff
@@ -52,63 +55,56 @@ class ProfilePage extends StatelessWidget {
...
+        return SingleChildScrollView(
+          scrollDirection: Axis.vertical,
+          child: Column(
+            children: <Widget>[
+              _buildProfileCard(model),
+              MenuList(
+                margin: _listContainerMargin,
+                itemList: _topMenuList,
+              ),
+              const SizedBox(height: 15.0),
```
Comment (Platton96) [inline ecommers/lib/ui/pages/profile_page.dart]: use insets

File: ecommers/lib/ui/pages/profile_page.dart (hunk 8 above)
Comment (Degorn) [reply ecommers/lib/ui/pages/profile_page.dart]: Value used

File: ecommers/lib/ui/pages/profile_page.dart (hunk 9 above)
Comment (Degorn) [reply ecommers/lib/ui/pages/profile_page.dart]: Done

**Review by Platton96 — APPROVED** [review]

**Pull request merged**